WordPress/gutenberg
# [Mobile] - Theme colors sync

`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2788
`WordPress iOS PR` -> https://github.com/wordpress-mobile/WordPress-iOS/pull/15258
`WordPress Android PR` -> https://github.com/wordpress-mobile/WordPress-Android/pull/13327

## Description
Fixes an issue with theme colors not [syncing correctly](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2637). 

~After some tests, it seems like the code removed in this PR was replacing the updated theme colors with the previous ones, this was introduced [here](https://github.com/WordPress/gutenberg/pull/23075) so I wonder if after the `useEditorFeature( 'color.palette' )` feature was introduced this is not needed anymore.~

**Update**: So I investigated a bit more and the issue was that passing down the colors/gradients as props were generating [another update](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/provider/index.native.js#L20) of the settings in the state triggered after the [theme colors callback](https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/provider/index.native.js#L132) hence replacing the new colors with the old ones.

[I updated it](https://github.com/WordPress/gutenberg/pull/26821/files#diff-f06faf4815cadf891d1ed891241c2e2bcaa2381e8a3043115d49fbdc1043d621R78) to set the initial colors using `updateSettings` instead of passing them as props to prevent them from replacing newer color updates in the state.

I went through the test cases of it and it looks like it's working as expected. @chipsnyder I'll add you as a reviewer because you'll have more insight related to this, am I missing something related to the theme colors that still requires this code?

## How has this been tested?
<details><summary>Steps to test updating the theme from the app</summary>

---
- Using the WordPress app
- Go to Themes and activate **Twenty Twenty**
- Create a post
- Add a cover block
- **Expect** to see the following colors:
<kbd><img src="https://user-images.githubusercontent.com/4885740/98666421-4dd21680-234d-11eb-835f-0f84e0437c6e.png" width="100"/></kbd>
- Save the post and close it
- Go to Themes and activate **Barnsbury**
- Open the previously created post
- **Expect** to see the following colors:
<kbd><img src="https://user-images.githubusercontent.com/4885740/98667305-86bebb00-234e-11eb-95ca-04eed5593136.png" width="100"/></kbd>

</details>

<details><summary>Steps to test updating the theme from the web</summary>

---
- Using the WordPress app
- Create a post
- Add a cover block (**don't select** an image/background color), save and close it
- Close the app _(leaving it in the background is fine)_
- On the web, go to your site and set the theme **Stow**
- Open the app
- Open the previously created post that has a Cover block
- **Expect** to see the following colors:
<kbd><img src="https://user-images.githubusercontent.com/4885740/98668956-ea49e800-2350-11eb-9bae-489c10d962d7.png" width="100"/></kbd>
- Close the post
- Close the app _(leaving it in the background is fine)_
- On the web, go to your site and set the theme **Brompton**
- Open the app
- Open the previously created post that has a Cover block
- **Expect** to see the following colors:
<kbd><img src="https://user-images.githubusercontent.com/4885740/98669295-81af3b00-2351-11eb-9558-db57d09a2d40.png" width="100"/></kbd>
</details>


<details><summary>Steps to test Editor Theme</summary>

---
Editor Theme - 1
- [ ] Default Colors - Check that default colors still load - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/editor-theme.md#tc001)
- [ ] Default Gradients - Check that default gradients still load - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/editor-theme.md#tc002)
- [ ] Custom Colors - Check that custom colors load in the editor - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/editor-theme.md#tc003)

Editor Theme - 2
- [ ] Custom Gradients - Check that custom gradients load in the editor - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/editor-theme.md#tc004)
- [ ] Offline Support - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/editor-theme.md#tc005)
</details>

## Types of changes
Bug fix

## Checklist:
